src/golang/lib/livy/types.go

```diff
@@ -0,0 +1,124 @@
+package livy
+
+type StatementState string
+
+type SessionState string
+
+type StatementOutputStatus string
+
+const (
+	NotStarted   SessionState = "not_started"
+	Starting     SessionState = "starting"
+	Idle         SessionState = "idle"
+	Busy         SessionState = "busy"
+	ShuttingDown SessionState = "shutting_down"
+	SessionError SessionState = "error"
+	Dead         SessionState = "dead"
+	Killed       SessionState = "killed"
+	Success      SessionState = "success"
+
+	Waiting        StatementState = "waiting"
+	Running        StatementState = "running"
+	Available      StatementState = "available"
+	StatementError StatementState = "error"
+	Cancelling     StatementState = "cancelling"
+
+	OK    StatementOutputStatus = "ok"
+	Error StatementOutputStatus = "error"
+)
+
+// Livy Session.
+type Session struct {
```

Comment:
Where are these statements come from? Looks like there's some public reference on how to define those?

Reply:
https://livy.incubator.apache.org/docs/latest/rest-api.html

Reply:
I see, let's ship this version and get the feature done first. I do feel we can separately explore the livy go package mentioned above, as it appear to have everything implemented